app/error/errors.js

```diff
@@ -19,6 +19,11 @@ export class HttpError extends Error {
  * Bad request
  */
 
+export class GenericLoginFailed extends HttpError {
```

Comment:
(nit) to keep consistent for error class-names, errors should be prefixed with `Error`. This just helps for clarity, so people can expect this error when they need to use it.